src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -249,7 +246,11 @@ private static Binder GetEnclosingBinder(CSharpSyntaxNode node, int position, Bi
                 {
                     binder = rootBinder.GetBinder(current);
                 }
-                else if (current.Kind() == SyntaxKind.ArrowExpressionClause && current.Parent?.Kind() == SyntaxKind.LocalFunctionStatement)
```

Comment:
Could this ever have been hit? I would be surprised if the `MemberSemanticModel` was called for statements inside a block.